lightyear/src/netconnection/server.rs

```diff
@@ -0,0 +1,26 @@
+use crate::_reexport::ReadWordBuffer;
+use crate::netconnection::ClientId;
+use crate::prelude::Io;
+use steamworks::SteamError;
+
+use anyhow::Result;
+
+/// Event emitted on each server update, to notify a new client connections or disconnections.
+pub struct ConnectionEvents {
+    pub connected: Vec<ClientId>,
+    pub disconnected: Vec<ClientId>,
+}
+
+pub trait NetServer {
+    /// Return the list of connected clients
+    fn connected_client_ids(&self) -> Vec<ClientId>;
```

Comment:
Preferably this wouldn't force an allocation if the user didn't need it, preferring to return a reference or iterator.

Unfortunately rust doesn't support `fn connected_client_ids(&self) -> impl Iterator<Item = ClientId>;` on a trait at the moment :(

An intermediate step would be returning [`Keys`](https://doc.rust-lang.org/std/collections/hash_map/struct.Keys.html) but it doesn't feel _great_

Not a huge thing, I doubt this will be called very often.